File: Java Pattern Programs/Wave Patterns/Pattern 17.java
 class Wave_17
 {
     public static void main(String args[])
     {
 
     int wave_height = 5; //change  value to increase or 
                    decrease the height of wave
     int wave_length = 4; //change  value to increase or 
                    decrease the length of wave
 
     int i;
     int j;
     int k;
     int is = 1; //inner space
     int os = 2; //outer space
 
     int inc = 1;
     int jump = (wave_height * 3) - (wave_height + 1);
 
     int num = wave_height;
 
      
 
 
     for (i = 1;i <= wave_height;i++)
     {
       for (j = 1;j <= wave_length;j++)
       {
         for (k = 1;k <= os;k++)
         {
          System.out.print(" ");
         }
          System.out.printf("%02d", num);
 
         for (k = 1;k <= is;k++)
         {
          System.out.print(" ");
         }
           num += inc;
 
          System.out.printf("%02d", num);
 
         for (k = 1;k <= os;k++)
         {
          System.out.print(" ");
         }
 
      System.out.print(" ");
     num += jump;
       }
 
 os=( (i + 1) != wave_height)?1:0;
         /* set value of os to 1 if i+1 is not equal to wave 
         height or 0 otherwise*/
 
 
         is = (i + 1 != wave_height)?3:5;
         /* set value of is to 3 if i+1 is not equal to wave 
         height or 5 otherwise*/
 
         num = wave_height - i;
         inc = inc + 2;
         jump -= 2;
 
         System.out.print(" \n");
     }
 
      
     }
 }

File: PythonPatternPrograms/WavePatterns/Pattern 17.py
wH = 5  # 1
wL = 4  # 2
insp = 1  # 3
ousp = 2  # 4
inc = 1
jump = (wH * 3) - (wH + 1)
num = wH

for x in range(1, wH + 1):
    for y in range(1, wL + 1):
        for z in range(1, ousp + 1):
            print(" ", end="")
        print("{:02d}".format(num), end="")
        for z in range(1, insp + 1):
            print(" ", end="")
        num += inc
        print("{:02d}".format(num), end="")
        for z in range(1, ousp + 1):
            print(" ", end="")
        print(" ", end="")
        num += jump

    ousp = (x + 1 != wH)  # 5

    if (x + 1 != wH):  # 6
        insp = 3
    else:
        insp = 5

    num = wH - x
    inc += 2
    jump -= 2
    print()

    """
1) wH - change value to increase/decrease the height of the wave
2) wL - change value to increase/decrease the length of the wave
3) inner space
4) outer space
5) set value of ousp to 1 if x+1 is not equal to wave height (wH) or 0 otherwise
6) set value of insp to 3 if x+1 is not equal to wave height (wH) or 5 otherwise
    """